common/v2/features/InteractWithContracts/components/Interact.tsx

```diff
@@ -187,9 +198,32 @@ export default function Interact(props: Props) {
   useEffect(() => {
     displayGeneratedForm(false);
     setWasContractInteracted(false);
+
+    if (areFieldsPopulatedFromUrl && networkAndAddressMatchURL && !wasAbiEditedManually) {
+      submitInteract();
+      const { function: functionFromUrl, input } = qs.parse(props.location.search);
+
+      const inputsArray: string[] = !input ? [] : Array.isArray(input) ? input : [input];
+
+      const inputsFromUrl = inputsArray.map(i => ({
+        name: i.includes(':') ? i.substr(0, i.indexOf(':')) : '',
+        value: i.includes(':') ? i.substr(i.indexOf(':') + 1, i.length) : ''
+      }));
```

Comment:
We should extract and test this logic along with L189 & 191.
Maybe a function that takes `props.location.search` and returns an object that is sanitised and can directly be used by the form?

Reply:
Made some tests. Name of the contract could be anything (also containing special chars - `<>()'`...) so I don't think we can do much to sanitise the query params. Parsed params are only used in the equality checks and shown in react components as inputs, so there shouldn't be security issues because React handles XSS attack for us.

Reply:
From what I understand the flow for `input` is:
`url -> location.search (react-router) -> parse (query-string) -> setInteractionDataFromURL --> then passing interactionDataFromURL as prop for GeneratedInteractionForm`

It appears that attacker controlled props are not escaped in React:
- https://stackoverflow.com/a/51852579/2057532
- https://medium.com/javascript-security/avoiding-xss-in-react-is-still-hard-d2b5c7ad9412

Though 'query-string' appears to handle escape attempts correctly.
Merging now to move forward.
Tagging @409H to see if this introduces a xss vulnerability: 

https://mycryptobuilds.com/pr/2920/#/interact-with-contracts?network=Ethereum&address=0x986EE2B944c42D017F52Af21c4c69B84DBeA35d8&function=allowance&input=0:0x986EE2B944c42D017F52Af21c4c69B84DBeA35d8&input=1:0x986EE2B944c42D017F52Af21c4c69B84DBeA35d8